File: ticket-mgt-api-java/src/main/java/com/ticketmgt/settings/service/PostmortemTemplateService.java
package com.ticketmgt.settings.service;

import com.ticketmgt.settings.dto.CreateTemplateRequest;
import com.ticketmgt.settings.dto.PostmortemTemplateDTO;
import com.ticketmgt.settings.entity.PostmortemTemplate;
import com.ticketmgt.settings.exception.ResourceNotFoundException;
import com.ticketmgt.settings.mapper.PostmortemTemplateMapper;
import com.ticketmgt.settings.repository.PostmortemTemplateRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class PostmortemTemplateService {

    private final PostmortemTemplateRepository templateRepository;
    private final PostmortemTemplateMapper templateMapper;

    @Transactional(readOnly = true)
    public List<PostmortemTemplateDTO> listTemplates() {
        return templateMapper.toDtoList(templateRepository.findAll());
    }

    @Transactional
    public void createTemplate(CreateTemplateRequest request) {
        PostmortemTemplate template = PostmortemTemplate.builder()
                .name(request.getName())
                .content(request.getContent())
                .build();
        templateRepository.save(template);
    }

    @Transactional
    public void updateTemplate(UUID id, CreateTemplateRequest request) {
        PostmortemTemplate template = templateRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Template not found"));
                
        template.setName(request.getName());
        template.setContent(request.getContent());
        templateRepository.save(template);
    }

    @Transactional
    public void deleteTemplate(UUID id) {
        PostmortemTemplate template = templateRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Template not found"));
        templateRepository.delete(template);
    }
}

File: ticket-mgt-app/src/services/postmortemTemplateService.ts
import apiClient from './apiClient';
import type {
  PostmortemTemplate,
  CreateTemplateRequest,
} from '../types/settings';

export const postmortemTemplateService = {
  /**
   * GET /postmortem-templates
   * Lists all postmortem templates for the organization.
   */
  listTemplates: async (): Promise<PostmortemTemplate[]> => {
    const { data } = await apiClient.get('/postmortem-templates');
    return data;
  },

  /**
   * POST /postmortem-templates
   * Creates a new postmortem template.
   */
  createTemplate: async (payload: CreateTemplateRequest): Promise<void> => {
    await apiClient.post('/postmortem-templates', payload);
  },

  /**
   * PUT /postmortem-templates/{id}
   * Updates an existing postmortem template.
   */
  updateTemplate: async (id: string, payload: CreateTemplateRequest): Promise<void> => {
    await apiClient.put(`/postmortem-templates/${id}`, payload);
  },

  /**
   * DELETE /postmortem-templates/{id}
   * Deletes a postmortem template.
   */
  deleteTemplate: async (id: string): Promise<void> => {
    await apiClient.delete(`/postmortem-templates/${id}`);
  },
};
